src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -401,13 +420,37 @@ void ClusterQueueHelper::afterPartitionPrimaryAssignment(
     }
 }
 
-void ClusterQueueHelper::assignQueue(const QueueContextSp& queueContext)
+void ClusterQueueHelper::assignQueueIfNeeded(
+    bmqp_ctrlmsg::Status* status,
+    const QueueContextSp& queueContext_sp)
+{
+    // executed by the cluster *DISPATCHER* thread
+
+    // PRECONDITIONS
+    BSLS_ASSERT_SAFE(
+        d_cluster_p->dispatcher()->inDispatcherThread(d_cluster_p));
+
+    BSLS_ASSERT_SAFE(queueContext_sp);
+
+    if (!isQueueAssigned(*queueContext_sp)) {
+        // Queue is not assigned to a partition; get it assigned.  If
+        // self is leader, it will assign it locally, if not it will
+        // send a request to the leader, etc.
+
+        assignQueue(status, queueContext_sp);
+    }
+}
+
+void ClusterQueueHelper::assignQueue(bmqp_ctrlmsg::Status* status,
+                                     const QueueContextSp& queueContext)
 {
     // executed by the cluster *DISPATCHER* thread
 
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(
         d_cluster_p->dispatcher()->inDispatcherThread(d_cluster_p));
+
+    BSLS_ASSERT_SAFE(queueContext);
     BSLS_ASSERT_SAFE(!isQueueAssigned(*queueContext));
```

Comment:
Is it necessary to have 2 functions now `assignQueue` / `assignQueueIfNeeded`?
We can have 1 function with early return
```
if (isQueueAssigned(*queueContext_sp)) {
    return;  // RETURN
}

// Queue is not assigned to a partition; get it assigned.  If
// self is leader, it will assign it locally, if not it will
// send a request to the leader, etc.
```

It allows to simplify class interface, reduce the number of lines and remove duplicating `ASSERT_SAFE` checks that check the same.